onedal/_device_offload.py

```diff
@@ -203,11 +204,15 @@ def wrapper_impl(obj, *args, **kwargs):
                 if dpnp_available and isinstance(args[0], dpnp.ndarray):
                     result = _convert_to_dpnp(result)
                 return result
-            input_array_api = getattr(data[0], "__array_namespace__", None)
-            if input_array_api:
-                input_array_api = input_array_api()
-                input_array_api_device = data[0].device
-                result = _asarray(result, input_array_api, device=input_array_api_device)
+            config = get_config()
+            if not ("transform_output" in config and config["transform_output"]):
```

Comment:
Will be refactored. Follow up ticket is created.

Reply:
Good catch though, ```transform_output``` is a tricky addition from sklearn 1.2